pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt

```diff
@@ -385,34 +345,25 @@ class KotlinCodeGenerator(
     }
 
     fun generateToStringMethod(): FunSpec {
-      val builder = FunSpec.builder("toString").addModifiers(KModifier.OVERRIDE).returns(STRING)
-
-      var builderSize = 50
-      val appendBuilder = CodeBlock.builder()
-      for (propertyName in allProperties.keys) {
-        builderSize += 50
-        appendBuilder.addStatement(
-          "appendProperty(builder, %S, this.%N)",
-          propertyName,
-          propertyName
-        )
-      }
-
-      builder
-        .addStatement("val builder = %T(%L)", STRING_BUILDER, builderSize)
+      return FunSpec.builder("toString")
+        .addModifiers(KModifier.OVERRIDE)
+        .returns(STRING)
         .addStatement(
-          // generate `::class.java.simpleName` instead of `::class.simpleName`
-          // to avoid making user code depend on kotlin-reflect
-          "builder.append(%T::class.java.simpleName).append(\" {\")",
-          kotlinPoetClassName
+          "return %P",
+          buildString {
+            append(pClass.simpleName)
+            append("(")
+            var isFirst = true
+            for (propertyName in allProperties.keys) {
+              if (isFirst) isFirst = false else append(", ")
+              append(propertyName)
+              append("=$")
+              append(propertyName)
```

Comment:
This needs to handle possibly quoted identifiers. Played around with it and this seems to do the trick:

```suggestion
              val propertyNameQuoted = CodeBlock.of("%N", propertyName).toString()
              if (propertyName == propertyNameQuoted) {
                append(propertyName)
              } else {
                append("{")
                append(propertyNameQuoted)
                append("}")
              }
```

Reply:
Good catch. I ended up using a `CodeBlock` for the entire interpolated string (as recommended in docs).